diy-omd.ino

```diff
@@ -85,6 +106,244 @@ void loop() {
     Serial.println("Temp C @ Index 0: " + String(temperatureSensors.getTempCByIndex(0))); // Get the first temperature.
   }
 
+  processIncomingBTData();
+  operateMotor();
+  processHallSensor();
+  logRPM();
+
+  if (testMillis > 500) {
+    test++;
+    transmitBTData("temp", (String)test);
+    testMillis = 0;
+  }
+}
+
+void transmitBTData(String key, String value) {
+  char str[50];
+  sprintf(str, "{\"%s\":\"%s\"}", key, value);
+
+  btSerial.write(str);
+  memset(str, 0, 50);
```

Comment:
If there is a reason to memset here, can you comment on why? This array should be freed automatically when the function ends, so I don't get why we'd need to zero it out first.